src/main/kotlin/com/learnspigot/bot/command/StartvoteCommand.kt

```diff
@@ -0,0 +1,123 @@
+import com.learnspigot.bot.manager.LeaderboardManager
+import dev.minn.jda.ktx.events.onCommand
+import dev.minn.jda.ktx.interactions.commands.restrict
+import dev.minn.jda.ktx.interactions.commands.upsertCommand
+import dev.minn.jda.ktx.messages.Embed
+import dev.morphia.Datastore
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Guild
+import net.dv8tion.jda.api.entities.Message
+import net.dv8tion.jda.api.entities.MessageEmbed
+import net.dv8tion.jda.api.entities.channel.forums.ForumTag
+import net.dv8tion.jda.api.entities.emoji.Emoji
+import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent
+import java.util.*
+import kotlin.concurrent.schedule
+
+class StartvoteCommand(
+    private val guild: Guild,
+    private val bot: JDA,
+    private val datastore: Datastore,
+    private val leaderboardManager: LeaderboardManager
+) {
+    fun startvoteCommand() {
+        guild.upsertCommand("startvote", "Start a vote for project or tutorial") {
+            restrict(guild = true)
+            bot.onCommand("startvote") {
+                if (guild.getMemberById(it.user.id)!!.roles.contains(guild.getRoleById(System.getenv("SUPPORT_ROLE_ID"))) && bot.getThreadChannelById(
+                        it.channel!!.id
+                    )?.parentChannel?.id != null && bot.getThreadChannelById(
+                        it.channel!!.id
+                    )!!.parentChannel.id == System.getenv("FOR_REVIEW_CHANNEL_ID")
+                ) {
+                    it.replyEmbed({
+                        title = "Success"
+                        description = "A vote has been started, results will be announced in 24 hours"
+                        color = EMBED_COLOR
+                    }).queue()
+                    val tutorialEmojis = mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣")
+                    val projectEmojis =
+                        mutableListOf("0️⃣", "1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️⃣", "7️⃣", "8️⃣", "9️⃣", "🔟")
+                    val type = bot.getThreadChannelById(it.channel!!.id)!!.appliedTags[0].name
+                    var message: Message? = null
+                    bot.getTextChannelById(System.getenv("VOTE_CHANNEL_ID"))!!
+                        .sendMessageEmbeds(voteEmbed(type, it, false)).queue { m ->
+                            message = m
+                            if (type == "Tutorial") tutorialEmojis.forEach { e ->
+                                m.addReaction(Emoji.fromUnicode(e)).queue()
+                            }
+                            else projectEmojis.forEach { e -> m.addReaction(Emoji.fromUnicode(e)).queue() }
+                        }
+                    Timer().schedule(86400000) {
+                        var sum = 0
+                        val usersReacted = mutableListOf<String>()
+                        val usersReaction = mutableListOf<Int>()
+                        if (type == "Tutorial") {
```

Comment:
I feel, in order for this to be more expandable, we use a when clause instead of if for this.